Wiki Updation #3 (#153)
NilPointer-Software/bdfd-wiki: src/guides/bdscript2.md

# BDScript 2

## Introduction

> **BDScript 2** is the default scripting language in-app _(as of October 2021)_. It has been created with intention of enhancing it's capabilities and fixing some of the problems the other versions had.

Reviewer: Move `in-app` in front of `scripting` to create `default in-app scripting language`
Also, should be `the problems previous versions had.`

```suggestion
**BDScript 2** is the default in-app scripting language _(as of October 2021)_. It has been created with intention of enhancing its capabilities and fixing some of the problems previous versions had.
```

The first edition of **BDScript** has one big issue, commands like `$sum[$sum[3;2];1]` didn't work. The reason it didn't work is because **BDScript** has a pre-defined order for executing functions.

In order to fix the issue, a new BDScript edition was developed called **BDScript Unstable**. It executes function in a command from bottom to top and from right to left.

> It fixed the issue but the new edtition had its own quirks which for some commands could be problematic. That's why **BDScript 2** comes in. This edition executes commands from top to bottom and from left to right *(basically, just the way you read most of the books)*.\

Reviewer: Should be `It fixes the issue, but the new edition has its quirks which could be problematic for some commands. That's where **BDScript 2** comes in.`
```suggestion
It fixes the issue, but the new edition has its quirks which could be problematic for some commands. That's where **BDScript 2** comes in. This edition executes commands from top to bottom and from left to right *(basically, just the way you read most of the books)*.\
```
